homework2/hw1.py

```diff
@@ -0,0 +1,122 @@
+"""
+Given a file containing text. Complete using only default collections:
+    1) Find 10 longest words consisting from largest amount of unique symbols
+    2) Find rarest symbol for document
+    3) Count every punctuation char
+    4) Count every non ascii char
+    5) Find most common non ascii char for document
+"""
+from typing import List
+from math import inf
+
+
+def get_longest_diverse_words(file_path: str) -> List[str]:
+    """
+    Find 10 longest words consisting from largest amount of unique symbols
+    :param file_path: name and path of text file
+    :return: list of the longest words
+    """
+    dict_length = {}
+    with open(file_path, encoding='unicode_escape', errors="ignore") as fi:
+        for line in fi:
+            line_list = line.split()
+            for word in line_list:
+                if dict_length.get(len(set(word))) is None:
+                    dict_length[len(set(word))] = [word]
```

Comment:
you count punctuation symbols as letters here. punctuation needs to be filtered too